nilearn/_utils/niimg_conversions.py

```diff
@@ -134,8 +161,11 @@ def concat_niimgs(niimgs, dtype=np.float32, accept_4d=False,
         A single image.
     """
 
+    if len(niimgs) == 0:
+        raise TypeError('Cannot concatenate empty objects')
```

Comment:
This may be a regression so I fixed (it was easy). As said in my comment, I don't make the code iterator friendly. This is a lot of work and I have no time to do it. Plus I prefer to segment things in smaller PRs.